pyazhpc/azhpc.py

```diff
@@ -465,43 +466,51 @@ def wrapped_print(indent, text, max_width=80):
 
         sys.exit(1)
 
-# create list of nodes from string 
 def _nodelist_expand(nodelist):
+    """Create list of nodes from string"""
+
     nodes = []
-    
+    resource_names = []
+
     # loop around resource
     for m in re.finditer(r"(?:,)?([^,^[]*(?:\[[^\]]*\])?)", nodelist):
         if len(m.groups()) > 0 and m.group(1) != "":
             nodestr = m.group(1)
-            
+
             # expand brackets
-            resource_name, brackets = re.search(r'([^[]*)\[?([\d\-\,]*)\]?', nodestr).groups(0)
+            resource, brackets = re.search(r'([^[]*)\[?([\d\-\,]*)\]?', nodestr).groups(0)
             if bool(brackets):
                 for part in brackets.split(","):
                     if "-" in part:
                         lo, hi = part.split("-")
                         assert len(lo) == 4, "expecting number width of 4"
                         assert len(hi) == 4, "expecting number width of 4"
                         for i in range(int(lo), int(hi) + 1):
-                            nodes.append(f"{resource_name}{i:04d}")
+                            nodes.append(f"{resource}{i:04d}")
                     else:
                         assert len(part) == 4, "expecting number width of 4"
-                        nodes.append(f"{resource_name}{part}")
+                        nodes.append(f"{resource}{part}")
+                resource_names.append(resource)
             else:
-                nodes.append(resource_name)
-    
-    return nodes
+                nodes.append(resource)
+                resource_names.append(resource[:-4])
+
+    # Remove trailing hyphen if used in node names to separate
```

Comment:
is there a case when this happens?  Azurehpc is currently coded to append a zero padded 4 digit number when creating multiple instances of a VM.

Reply:
I choose to add a hyphen in the node names to separate the resource name and the node index. The reason being that when the resource name ends with one or more digits (e.g. `d64dv4`), then when resuming multiple nodes, Slurm returns a node list in compact form and the digits in the resource name are added into the indexes (e.g. `d64dv[4001-4005]`), which causes `_nodelist_expand` to fail since the index has more than four digits.

Reply:
Decided to keep hyphen separation in offline discussion. Resolving conversation.